### PR TITLE
chore!: refactore store key constructors

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

Follow up on https://github.com/umee-network/umee/pull/1573
Cleanup store key prefix functions.
 
---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by rbajollari — APPROVED** [review]
👍

**Review by gsk967 — APPROVED** [review]
LGTM

File: x/leverage/types/keys.go [hunk 1]
```diff
@@ -34,36 +35,30 @@ var (
...
 }
 
 // CreateAdjustedBorrowKey returns a KVStore key for getting and setting an
 // adjusted borrow for a denom and borrower address.
 func CreateAdjustedBorrowKey(borrowerAddr sdk.AccAddress, tokenDenom string) []byte {
-	// borrowprefix | lengthprefixed(borrowerAddr) | denom | 0x00
+	// borrowprefix | lengthprefixed(borrowerAddr) | denom | 0x00 for null-termination
 	key := CreateAdjustedBorrowKeyNoDenom(borrowerAddr)
 	key = append(key, []byte(tokenDenom)...)
 	return append(key, 0) // append 0 for null-termination
```
Comment (toteki) [inline x/leverage/types/keys.go:48]:
```suggestion
	return util.ConcatBytes(1, CreateAdjustedBorrowKeyNoDenom(borrowerAddr),[]byte(tokenDenom))
```

File: x/leverage/types/keys.go [hunk 2]
```diff
@@ -34,36 +35,30 @@ var (
...
 }
 
 // CreateCollateralAmountKey returns a KVStore key for getting and setting the amount of
 // collateral stored for a user in a given denom.
 func CreateCollateralAmountKey(addr sdk.AccAddress, uTokenDenom string) []byte {
-	// collateralPrefix | lengthprefixed(addr) | denom | 0x00
+	// collateralPrefix | lengthprefixed(addr) | denom | 0x00 for null-termination
 	key := CreateCollateralAmountKeyNoDenom(addr)
 	key = append(key, []byte(uTokenDenom)...)
 	return append(key, 0) // append 0 for null-termination
```
Comment (toteki) [inline x/leverage/types/keys.go:64]:
```suggestion
	return util.ConcatBytes(1, CreateCollateralAmountKeyNoDenom(addr),[]byte(uTokenDenom))
```

**Review by toteki — APPROVED** [review]
Looks good - also I added some leverage `key_test.go` cases.

File: x/oracle/types/keys.go [hunk 3]
```diff
@@ -30,61 +31,52 @@ var (
...
+	return util.ConcatBytes(1, KeyPrefixExchangeRate, []byte(denom))
 }
 
-// GetFeederDelegationKey - stored by *Validator* address
-func GetFeederDelegationKey(v sdk.ValAddress) (key []byte) {
-	key = append(key, KeyPrefixFeederDelegation...)
-	return append(key, address.MustLengthPrefix(v)...)
+// KeyFeederDelegation - stored by *Validator* address
+func KeyFeederDelegation(v sdk.ValAddress) []byte {
+	// note: length prefix is not needed here!
```
Comment (toteki) [inline x/oracle/types/keys.go:42]:
```suggestion
```

**Comment by robert-zaremba** [comment]
> Looks good - also I added some leverage key_test.go cases.

Thank you.

File: x/oracle/types/keys.go (hunk 3 above)
Comment (robert-zaremba) [reply x/oracle/types/keys.go:42]: I will create an issue instead of this comment.

**Pull request merged**